is_iam_enabled.go

```diff
@@ -0,0 +1,74 @@
+package alks
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+)
+
+type IsIamEnabledRequest struct {
+	Account string `json:"account,omitempty"`
+	Role    string `json:"role,omitempty"`
+	RoleArn string `json:"roleArn,omitempty"`
+}
+
+// IsIamEnabledResponse is used to represent a role that's IAM active or not.
+type IsIamEnabledResponse struct {
+	BaseResponse
+	Account    string `json:"account,omitempty"`
+	Role       string `json:"role,omitempty"`
+	RoleArn    string `json:"roleArn"`
+	IamEnabled bool   `json:"iamEnabled"`
+}
+
+// IsIamEnabled will check if a MI, AccountDetails, or STS assumed role is IAM active or not.
+func (c *Client) IsIamEnabled(roleArn string) (*IsIamEnabledResponse, error) {
+
+	if len(roleArn) > 1 {
+		log.Printf("[INFO] Is IAM enabled for MI: %s", roleArn)
+	} else {
+		log.Printf("[INFO] Is IAM enabled for: %s", c.AccountDetails.Account)
+		log.Printf("[INFO] Is IAM enabled for: %s", c.AccountDetails.Role)
+	}
+
+	iam := IsIamEnabledRequest{
+		c.AccountDetails.Account,
+		c.AccountDetails.Role,
```

Comment:
I believe this can be simplified to this, but you might need to double check.
```suggestion
		c.AccountDetails,
```